src/argus/notificationprofile/serializers.py

```diff
@@ -137,27 +137,34 @@ class Meta:
         ]
 
     def validate(self, attrs: dict):
-        if self.instance and "media" in attrs.keys() and not attrs["media"].slug == self.instance.media.slug:
-            raise serializers.ValidationError("Media cannot be updated, only settings.")
-        if "settings" in attrs.keys():
-            if not isinstance(attrs["settings"], dict):
-                raise serializers.ValidationError("Settings has to be a dictionary.")
-            if self.instance:
-                medium = api_safely_get_medium_object(self.instance.media.slug)
-            else:
-                medium = api_safely_get_medium_object(attrs["media"].slug)
-            attrs["settings"] = medium.validate(self, attrs, self.context["request"].user)
+        settings = attrs.get("settings", None)
+        if not settings:
+            raise serializers.ValidationError("Settings cannot be empty")
+
+        if self.instance:
+            medium = api_safely_get_medium_object(self.instance.media.slug)
+            # A PATCH need not contain the media key
+            if attrs.get("media", None) != self.instance.media:
```

Comment:
Also here: If we have PATCH and no media in the attrs, then this will fail since `None != EmailMedium` for example

```suggestion
            if "media" in attrs.keys() and attrs["media"] != self.instance.media:
```

Reply:
Because we always have an instance when we patch we can use the medium of the instance, which a late commit does.

Reply:
The point of this code is to prevent changing of media after all.